Dockerfile

```diff
@@ -13,7 +13,6 @@ RUN cd /root/ \
  && /root/emsdk/emsdk activate latest \
  && source /root/emsdk/emsdk_env.sh --build=Release \
  && emcc hello_world.cpp \
- && rm ~/.emscripten \
```

Comment:
Is the intent of this Dockerfile to only have `latest-upstream` active at the end of this whole script, and nothing else? If so, then the line `&& /root/emsdk/emsdk activate latest \` looks redundant (and the `emsdk_env` line and compilation check should probably move down to after `activate latest-upstream` has been applied?)

Reply:
This checks both paths - it compiles hello world with fastcomp, then with the wasm backend. So it's also checking switching between them.